### PR TITLE
Abstract away remaining ebiten references

### PR DESCRIPTION
**Pull request by carrelld**
https://github.com/OpenDiablo2/OpenDiablo2/issues/337

I'm not sure how correct any of this is, so please give me some feedback so I can make necessary changes.

### CODE REVIEW

File: d2core/d2ui/d2ui.go [hunk 1]
```diff
@@ -17,122 +19,105 @@ const (
...
+
+	d2input.BindHandler(&singleton) // TODO bind here or at the calling code
 }
 
 // Reset resets the state of the UI manager. Typically called for new screens
 func Reset() {
-	widgets = make([]Widget, 0)
-	pressedIndex = -1
-	waitForLeftMouseUp = true
+	singleton = UI{}
```
Comment (carrelld) [inline d2core/d2ui/d2ui.go:45]: Given that we BindHandler on the pointer to the singleton above, we should probably reset fields (like the original code did) or UnbindHandler and then BindHandler as part of this Reset method

File: d2core/d2input/input_manager.go [hunk 2]
```diff
@@ -114,8 +124,8 @@ func (im *inputManager) advance(elapsed float64) error {
 		})
 	}
 
-	for button := ebiten.MouseButtonLeft; button < ebiten.MouseButtonMiddle; button++ {
-		if inpututil.IsMouseButtonJustPressed(button) {
+	for button := mouseButtonMin; button < mouseButtonMax; button++ {
```
Comment (nicholas-eden) [inline d2core/d2input/input_manager.go:127]: I don't know about `mouseButtonMin` and `mouseButtonMax`.  Its not really clear what they refer to.  We might support 5+ button mice, so assigning to the right button doesn't seem right.

File: d2core/d2ui/scrollbar.go [hunk 3]
```diff
@@ -51,8 +51,8 @@ func (v Scrollbar) getBarPosition() int {
 	return int((float32(v.currentOffset) / float32(v.maxOffset)) * float32(v.height-30))
 }
 
-func (v *Scrollbar) Activate() {
-	_, my := ebiten.CursorPosition()
+func (v *Scrollbar) OnMouseMove(event d2input.MouseMoveEvent) bool {
```
Comment (carrelld) [inline d2core/d2ui/scrollbar.go:54]: I haven't tested this change yet

File: d2core/d2input/input_manager.go [hunk 4]
```diff
@@ -26,7 +23,20 @@ func (lel handlerEntryList) Less(i, j int) bool {
...
+	IsKeyJustPressed(key Key) bool
+	IsKeyJustReleased(key Key) bool
+	IsMouseButtonPressed(button MouseButton) bool
+	IsMouseButtonJustPressed(button MouseButton) bool
+	IsMouseButtonJustReleased(button MouseButton) bool
+	KeyPressDuration(key Key) int
+}
+
 type inputManager struct {
+	in      InputService
```
Comment (essial) [inline d2core/d2input/input_manager.go:39]: please call this variable `inputerService`

File: d2core/d2input/input_manager.go (hunk 2 above)
Comment (essial) [reply d2core/d2input/input_manager.go:127]: I'm OK with this, we haven't really defined key binding support yet.

File: d2core/d2ui/scrollbar.go (hunk 3 above)
Comment (essial) [reply d2core/d2ui/scrollbar.go:54]: Just create a bunch of characters and it should activate the scrollbar.

**Comment by essial** [comment]
Review the comments by us above, also don't forget to add your name to CONTRIBUTORS file. Also please make sure you go through all the screens and make sure everything still works. Things like pressing on a button, moving off and release shouldn't active the button. That kind of stuff. After that I'll accept the merge!

**Comment by carrelld** [comment]
@essial All ready to go!

**Comment by essial** [comment]
Coolio. Let me know. Also have a conflict on contributors but that should be easy for you to resolve.

**Comment by carrelld** [comment]
@essial The change I made seems to have fixed it and not caused anything else to break. I'm ready to get this thing away from me now.

**Comment by essial** [comment]
> @essial The change I made seems to have fixed it and not caused anything else to break. I'm ready to get this thing away from me now.

Haha thanks!

**Pull request merged**